### PR TITLE
Handle invalid dates

### LINKED ISSUE #1294: Handle invalid dates:
screen shot 2016-10-07 at 1 31 27 pm

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

closes #1294 

#### What's this PR do?

This deals with invalid dates in the `CourseDescription` and `CourseAction` components, basically by testing to see if the date is valid before we try to format and display it.

#### How should this be manually tested?

What I did was alter the `getDashboard`, `getProgramEnrollments` and `getCoursePrices` functions in `api.js` to return their respective constants (from `constants.js`). Then you can alter relevant courseRuns in order to make invalid dates, to test that every case is covered. A good string to use is `'1999-13-92'`.

### CODE REVIEW

File: static/js/components/dashboard/CourseAction_test.js [hunk 1]
```diff
@@ -117,6 +117,17 @@ describe('CourseAction', () => {
     assertCheckoutButton(elements.button, firstRun.course_id);
   });
 
+  it('hides an invalid date with STATUS_OFFERED', () => {
+    let course = findCourse(course => (
+      course.runs.length > 0 &&
+      course.runs[0].status === STATUS_OFFERED
+    ));
+    alterFirstRun(course, {enrollment_start_date: '1999-13-92'});
```
Comment (noisecapella) [inline static/js/components/dashboard/CourseAction_test.js]: `alterFirstRun` will modify `course` so you'll need to clone it first. Can you change `findCourse` to `findAndCloneCourse` above?

File: static/js/util/date.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import R from 'ramda';
+
+export const ifValidDate = R.curry((def, fn, date) => (
```
Comment (noisecapella) [inline static/js/util/date.js]: Can you write a test for this in `date_test.js`?

File: static/js/components/dashboard/CourseAction_test.js (hunk 1 above)
Comment (noisecapella) [inline static/js/components/dashboard/CourseAction_test.js]: This also needs to be done in the other two tests added below

File: static/js/util/date.js (hunk 2 above)
Comment (aliceriot) [inline static/js/util/date.js]: oh yeah I meant to, just slipped my mind :)

**Comment by aliceriot** [comment]
@noisecapella comments addressed!

**Comment by noisecapella** [comment]
Looks good :+1:

**Pull request merged**